### PR TITLE
Verify that component namespace and name combinations are unique

### LINKED ISSUE #684: [FEATURE] Add check in `viash ns ...` to check namespace/component_name uniqueness
### Feature summary

Verify that all picked up components have an unique namespace & component_name combination.

### Feature description

It is currently possible to define 2 or more files with the same namespace & component names.
`src/ns1/comp.vsh.yml` and `src/ns2/comp.vsh.yml`
both containing:
```
functionality:
  name: comp
  namespace: ns
```

### Why is this feature beneficial?

Prevents collisions down the road.

### Alternatives considered

_No response_

### Possible solution

_No response_

### Confirmation

- [X] I have searched the existing issues to make sure this is not a duplicate request.
- [X] I have provided clear and concise information about the feature.

### PR DESCRIPTION
**Pull request by Grifs**
## Describe your changes

Verify that component namespace and name combinations are unique

## Related issue(s)

Closes #684 

## Type of Change

- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] New functionality (non-breaking change which adds functionality)
- [ ] Major change (non-breaking change which modifies existing functionality)
- [ ] Minor change (non-breaking change which does not modify existing functionality)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] This change requires a documentation update

## Checklist

Requirements:

- [x] I have read the [CONTRIBUTING](CONTRIBUTING.md) doc.
- [x] I have performed a self-review of my code by checking the "Changed Files" tab.
- [x] My code follows the code style of this project.

Tests:

- [x] I have added tests that prove my fix is effective or that my feature works.
- [x] New and existing unit tests pass locally with my changes.

Documentation:

- [x] Proposed changes are described in the CHANGELOG.md.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: src/main/scala/io/viash/config/Config.scala [hunk 1]
```diff
@@ -365,6 +365,16 @@ object Config extends Logging {
 
     val allConfigs = allConfigsWithStdOutErr.collect({ case Left((config, _, _)) if config.functionality.isEnabled => config })
 
+    // Verify that all configs except the disabled ones are unique
+    // Even configs disabled by the query should be unique as they might be picked up as a dependency
+    // Only allowed exception are configs where the status is set to disabled
+    val uniqueConfigs = allConfigs.groupBy(config => config.functionality.namespace.map(_ + "/").getOrElse("") + config.functionality.name)
```
Comment (rcannood) [inline src/main/scala/io/viash/config/Config.scala:371]: Perhaps you could also add organisation to this id. Wdyt?

Comment (Grifs) [reply src/main/scala/io/viash/config/Config.scala:371]: But the organisation isn't used in the output path, so collisions could still happen then.
However, this reminds me that I wanted to look into using `ViashNamespace.targetOutputPath(...)` as the key generator - as that is what determines what would result in collisions and means that once we have more flexibility in the folder name strategy as previously discussed, we're halfway there in solving that too.

Comment (rcannood) [reply src/main/scala/io/viash/config/Config.scala]: I see -- good point.

Should we even consider the use-case where one might use multiple organisations in one project -- perhaps not?

Comment (Grifs) [reply src/main/scala/io/viash/config/Config.scala]: The one I see most likely to happen is a project that sets the organisation but with one or more components where the organisation is not set. In that regard it's still 2 different values for the organisation.
In itself it doesn't really hurt much but it's probably an error.

If we want a check for that, I'd suggest displaying a warning and adding it to Viash 0.9 (or later).

**Review by rcannood — APPROVED** [review]
LGTM!

**Pull request merged**